workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java

```diff
@@ -172,6 +185,26 @@ public synchronized WorkFlowExecution updateWorkFlow(WorkFlowExecution workFlowE
 		return workFlowRepository.save(workFlowExecution);
 	}
 
+	@Override
+	public List<WorkFlowResponseDTO> getWorkFlowsByProjectId(UUID projectId) {
+		List<ProjectResponseDTO> projects = projectId == null
+				? projectService.findProjectsByUserName(SecurityUtils.getUsername())
+				: List.of(projectService.getProjectByIdAndUserName(projectId, SecurityUtils.getUsername()));
+		return projects.stream().flatMap(project -> workFlowRepository.findAllByProjectId(project.getId()).stream()
+				.filter(workFlowExecution -> workFlowExecution.getMainWorkFlowExecution() == null)
```

Comment:
a lot is going on in this method, but in terms of readability, I find its readability being somewhat complex.
do you think it can be refactored a bit?

Reply:
ok, I will move out the WorkFlowResponseDTO builder